### PR TITLE
Devel/revpi 2756/collect source modification

### PR DESCRIPTION
**Pull request by zhan-kunbus**

### CODE REVIEW

**Comment by nbuchwitz** [comment]
Please also remove the following:
https://github.com/RevolutionPi/imagebakery/blob/916746fd69a0d59b69467658a4f56fad4b9ec201/collect_sources.sh#L79-L80
https://github.com/RevolutionPi/imagebakery/blob/916746fd69a0d59b69467658a4f56fad4b9ec201/collect_sources.sh#L55 (wiringpi)

Further could you please check if we can download the node red sources from https://github.com/node-red/
Same applies to nodes and server from erminas:
https://github.com/erminas/noderedrevpinodes-server
https://github.com/erminas/noderedrevpinodes-server

**Comment by zhan-kunbus** [comment]
> Please also remove the following:
> 
> https://github.com/RevolutionPi/imagebakery/blob/916746fd69a0d59b69467658a4f56fad4b9ec201/collect_sources.sh#L79-L80
> 
> 
> https://github.com/RevolutionPi/imagebakery/blob/916746fd69a0d59b69467658a4f56fad4b9ec201/collect_sources.sh#L55
> 
> (wiringpi)

done

> Further could you please check if we can download the node red sources from https://github.com/node-red/ 

also the source package of nodejs included ?

> Same applies to nodes and server from erminas: https://github.com/erminas/noderedrevpinodes-server https://github.com/erminas/noderedrevpinodes-server

The packages are installed from revpi repo, and deb source package of this packages can be gotten from there, wouldn't it better to use it instead of erminas?
`pi@RevPi0110:~ $ apt source noderedrevpinodes-server
Reading package lists... Done
NOTICE: 'noderedrevpinodes-server' packaging is maintained in the 'Git' version control system at:
https://github.com/RevolutionPi/noderedrevpinodes-server.git
Please use:
git clone https://github.com/RevolutionPi/noderedrevpinodes-server.git
to retrieve the latest (possibly unreleased) updates to the package.
Need to get 18.3 kB of source archives.
Get:1 http://packages.revolutionpi.de buster/main noderedrevpinodes-server 1.0.4-1 (diff) [3,660 B]
Get:2 http://packages.revolutionpi.de buster/main noderedrevpinodes-server 1.0.4-1 (dsc) [2,055 B]
Get:3 http://packages.revolutionpi.de buster/main noderedrevpinodes-server 1.0.4-1 (tar) [12.6 kB]
Fetched 18.3 kB in 0s (96.3 kB/s)
dpkg-source: info: extracting noderedrevpinodes-server in noderedrevpinodes-server-1.0.4
dpkg-source: info: unpacking noderedrevpinodes-server_1.0.4.orig.tar.gz
dpkg-source: info: unpacking noderedrevpinodes-server_1.0.4-1.debian.tar.xz
dpkg-source: info: using patch list from debian/patches/series
dpkg-source: info: applying 0001-Remove-upstream-package-build-scripts.patch
`

**Comment by nbuchwitz** [comment]
1) NodeRed sadly doesn't provide sources packages. Therefore our only option is to download the sources from GitHub
2) Please use our packages for the erminas stuff as this is installed in our image. There are minor issues with the packaging of Erminas, so we decided to package it by ourself

**Comment by zhan-kunbus** [comment]
the node-red can be gotten from "https://github.com/node-red/node-red/archive//master.tar.gz" 
but it seems there are lots of "packages" named "node-*" installed by npm, which have no source package for them.

**Review by nbuchwitz — CHANGES_REQUESTED** [review]

File: collect_sources.sh [hunk 1]
```diff
@@ -68,22 +68,16 @@ dpkg-query --admindir $APTROOT/var/lib/dpkg -W		\
...
 	-f='${source:Version}' raspberrypi-kernel || true)
 knl_tag="raspberrypi-kernel_$knl_version"
+knl_tag=$(sed 's/:/%25/g' <<< "$knl_tag")
 wget -O "linux-$knl_version.tar.gz" "https://github.com/RevolutionPi/linux/archive/refs/tags/$knl_tag.tar.gz"
 wget -O "piControl-$knl_version.tar.gz" "https://github.com/RevolutionPi/piControl/archive/refs/tags/$knl_tag.tar.gz"
 wget -O IODeviceExample.tar.gz https://github.com/RevolutionPi/IODeviceExample/archive/master.tar.gz
-wget -O python-snap7.tar.gz https://github.com/RevolutionPi/python-snap7/archive/master.tar.gz
-wget -O snap7-debian.tar.gz https://github.com/RevolutionPi/snap7-debian/archive/master.tar.gz
 wget -O python3-revpimodio2.tar.gz https://github.com/naruxde/revpimodio2/archive/master.tar.gz
+wget -O node-red.tar.gz https://github.com/node-red/node-red/archive//master.tar.gz
```
[the remark this answers is not in the document]
Comment (zhan-kunbus) [reply collect_sources.sh]: the installed version of node-red can be downloaded now.

And for the other packages of npm,  would it be fine to make just a tarball from /usr/lib/node_modules ? @nbuchwitz

[the remark this answers is not in the document]
Comment (nbuchwitz) [reply collect_sources.sh]: Sounds reasonable, please do so.

**Review by nbuchwitz — APPROVED** [review]
LGTM

**Pull request merged**